recipe/thirdparty/providers/linkedin.go

```diff
@@ -0,0 +1,113 @@
+package providers
+
+import (
+	"errors"
+
+	"github.com/supertokens/supertokens-golang/recipe/thirdparty/tpmodels"
+	"github.com/supertokens/supertokens-golang/supertokens"
+)
+
+const linkedinID = "linkedin"
+
+func Linkedin(input tpmodels.ProviderInput) tpmodels.TypeProvider {
+	if input.ThirdPartyID == "" {
+		input.ThirdPartyID = linkedinID
+	}
+
+	if input.Config.AuthorizationEndpoint == "" {
+		input.Config.AuthorizationEndpoint = "https://www.linkedin.com/oauth/v2/authorization"
+	}
+
+	if input.Config.TokenEndpoint == "" {
+		input.Config.TokenEndpoint = "https://www.linkedin.com/oauth/v2/accessToken"
+	}
+
+	if input.Config.UserInfoMap.FromUserInfoAPI.UserId == "" {
+		input.Config.UserInfoMap.FromUserInfoAPI.UserId = "id"
+	}
+	if input.Config.UserInfoMap.FromUserInfoAPI.Email == "" {
+		input.Config.UserInfoMap.FromUserInfoAPI.Email = "email"
+	}
+	if input.Config.UserInfoMap.FromUserInfoAPI.EmailVerified == "" {
+		input.Config.UserInfoMap.FromUserInfoAPI.EmailVerified = "email_verified"
+	}
+
+	oOverride := input.Override
+
+	input.Override = func(provider *tpmodels.TypeProvider) *tpmodels.TypeProvider {
+		oGetConfig := provider.GetConfigForClientType
+		provider.GetConfigForClientType = func(clientType *string, input tpmodels.ProviderConfig, userContext supertokens.UserContext) (tpmodels.ProviderConfigForClientType, error) {
+			config, err := oGetConfig(clientType, input, userContext)
+			if err != nil {
+				return tpmodels.ProviderConfigForClientType{}, err
+			}
+
+			if len(config.Scope) == 0 {
+				config.Scope = []string{"r_emailaddress", "r_liteprofile"}
+			}
+
+			return config, err
+		}
+
```

Comment:
See this issue: https://github.com/supertokens/supertokens-node/issues/291

Reply:
<img width="578" alt="image" src="https://user-images.githubusercontent.com/650429/202104501-457fc819-f444-40d7-bed3-712a6d11635b.png">

We are doing the x-www-form-urlencoded content type only by default, that's why Linked in works as expected. If some provider needs to post a JSON body, ExchangeCode function will have to be overridden.